src/Widget/Field.php

```diff
@@ -0,0 +1,916 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Form\Widget;
+
+use InvalidArgumentException;
+use Yiisoft\Arrays\ArrayHelper;
+use Yiisoft\Factory\Exceptions\InvalidConfigException;
+use Yiisoft\Form\FormModelInterface;
+use Yiisoft\Form\Helper\HtmlForm;
+use Yiisoft\Html\Html;
+use Yiisoft\Widget\Widget;
+
+use function array_merge;
+
+/**
+ * Renders the field widget along with label, error tag and hint tag (if any) according to template.
+ */
+final class Field extends Widget
```

Comment:
There are many public methods that aren't used anywhere. I'm not sure all of them should be public.

Reply:
// create forms
```php
public function label(bool $enabledLabel = true, array $options = [], ?string $label = null): self
public function error(array $options = []): self
public function hint(?string $content = null, bool $typeHint = true, array $options = []): self
public function input(string $type, array $options = []): self
public function textInput(array $options = []): self
public function hiddenInput(array $options = []): self
public function passwordInput(array $options = []): self
public function fileInput(array $options = []): self
public function textArea(array $options = []): self
public function radio(array $options = [], bool $enclosedByLabel = true): self
public function checkbox(array $options = [], bool $enclosedByLabel = true): self
public function dropDownList(array $items, array $options = []): self
public function listBox(array $items, array $options = []): self
public function checkboxList(array $items, array $options = []): self
public function radioList(array $items, array $options = []): self
private function addInputId(): string
```

// config factory init
```php
public function config(FormModelInterface $data, string $attribute, array $options = []): self
public function ariaAttribute(bool $value): self
public function errorCssClass(string $value): self
public function errorSummaryCssClass(string $value): self
public function inputCssClass(string $value): self
public function requiredCssClass(string $value): self
public function successCssClass(string $value): self
public function template(string $value): self
public function validatingCssClass(string $value): self
public function validationStateOn(string $value): self
public function errorOptions(array $options = []): void
public function hintOptions(array $options = []): void
public function inputOptions(array $options = []): void
public function labelOptions(array $options = []): void
```

// privates.
```php
private function renderBegin(): string
private function renderEnd(): string
private function addErrorCssClassToContainer(): void
private function addErrorCssClassToInput(): void
private function setInputRole(array $options = []): void
private function skipForInLabel(): void
private function setForInLabel(array $options = []): void
private function setAriaAttributes(array $options = []): void
```

Reply:
so they are perfect.